File: data_structure/Stack.py
class Stack:

    def __init__(self):
        self.items = []

    def push(self, item):
        self.items.append(item)

    def pop(self):
        if len(self.items) > 0:
            print(self.items.pop())
        else:
            print(-1)

    def stack_size(self):
        print(len(self.items))

    def is_empty(self):
        if len(self.items) < 0:
            print(1)
        else:
            print(0)

    def print_top(self):
        if len(self.items) > 0:
            print(self.items[-1])
        else:
            print(-1)
    
 

File: data_structure/Stack.cpp
#include <iostream>
#define SIZE 100

using namespace std;

int Stack[SIZE];
int Top = -1;

void push(int data){

	if (Top == 99){
		cout << "This stack is full." << endl;
	}

	Top++;
	Stack[Top] = data;

	return;
}

int pop(){
	if (Top == -1){
		cout << "This stack is empty." << endl;
		return 0;
	}

	return Stack[Top];
}

int isempty(){
	if(Top == -1){
		return 1;
	}
	return 0;
}


int main(){

	push(10);
	push(5);
	cout << isempty() << endl;
	cout << pop();

	return 0;	
}